modules/sdk/project-templates/project-templates/src/main/java/com/liferay/project/templates/internal/util/ProjectTemplatesUtil.java

```diff
@@ -30,43 +37,139 @@
  */
 public class ProjectTemplatesUtil {
 
-	public static File getArchetypeFile(String artifactId, File file)
-		throws IOException {
+	public static File getArchetypeFile(String artifactId) throws IOException {
+		if (_archetypeFiles.containsKey(artifactId)) {
+			return _archetypeFiles.get(artifactId);
+		}
 
-		try (JarFile jarFile = new JarFile(file)) {
-			Enumeration<JarEntry> enumeration = jarFile.entries();
+		Properties projectTemplateJarVersions = getProjectTemplateJarVersions();
```

Comment:
Variable names usually end with the class for easier readability (i.e. `projectTemplateJarVersionsProperties`)